builtin/credential/approle/path_role.go

```diff
@@ -2014,10 +2019,16 @@ func (b *backend) pathRolePeriodRead(ctx context.Context, req *logical.Request,
 		return nil, nil
 	}
 
+	d := map[string]interface{}{
+		"token_period": role.TokenPeriod / time.Second,
+	}
+
+	if role.Period > 0 {
+		d["period"] = role.Period / time.Second
+	}
+
 	return &logical.Response{
-		Data: map[string]interface{}{
-			"period": role.Period / time.Second,
-		},
+		Data: d,
```

Comment:
Do we want to add a warning that both indicate the same and one of them is deprecated (or tell that it is there for backwards compatibility)?

Reply:
It's in pathhelp via tokenutil.DeprecationText